src/uu/date/src/date.rs

```diff
@@ -103,6 +107,75 @@ enum Iso8601Format {
     Ns,
 }
 
+// Enum for system clock resolution
+enum TimeResolution {}
+const RESOLUTION_DIVISOR: f32 = 1_000_000_000.0;
+
+impl TimeResolution {
+    fn generate() -> f32 {
+        // Getting time from system
+        let time = std::time::SystemTime::now();
+
+        let duration = time
+            .duration_since(std::time::UNIX_EPOCH)
+            .unwrap_or_default();
+
+        let secs: u64 = duration.as_secs();
+        let nsecs: u32 = duration.subsec_nanos();
+
+        let mut ts: timespec = timespec {
+            tv_sec: secs as c_long,
+            tv_nsec: nsecs as c_long,
+        };
+
+        let resolution: f32 = match Self::clock_type(&mut ts) {
+            Some(CLOCK_REALTIME) => {
+                let mut res_ts = ts;
+                unsafe { libc::clock_getres(CLOCK_REALTIME, &mut res_ts) };
+                let resolution_secs: f32 =
+                    res_ts.tv_sec as f32 + res_ts.tv_nsec as f32 / RESOLUTION_DIVISOR;
+                resolution_secs
+            }
+            Some(CLOCK_MONOTONIC) => {
+                let mut res_ts: timespec = ts;
+                unsafe { libc::clock_getres(CLOCK_MONOTONIC, &mut res_ts) };
+                let resolution_secs: f32 =
+                    res_ts.tv_sec as f32 + res_ts.tv_nsec as f32 / RESOLUTION_DIVISOR;
+                resolution_secs
+            }
+            _ => {
+                // We 'create' a resolution here based on how many digits we get for resolution_secs
+                let resolution_secs: f32 = ts.tv_nsec as f32 / RESOLUTION_DIVISOR;
+                let resolution_string: &String = &format!("{}", resolution_secs);
+                // We take 3 numbers off to add the '0.' for the decimal and add '1' to the end.
+                let resolution_length: u8 = (resolution_string.len() - 3).try_into().unwrap_or(1);
+                let mut resolution_final: String = String::new();
+                resolution_final.push_str("0.");
+                let mut i: u8 = 0;
+                while i <= resolution_length {
+                    resolution_final.push('0');
+                    i += 1;
+                }
+                resolution_final.push('1');
+
+                resolution_final.parse::<f32>().unwrap_or(0.1)
```

Comment:
```suggestion
                (0.1).powi(resolution_length)
```

Also, I'm unhappy that this "guessing" has to happen at all. For example, on a system with a 1-nanosecond resolution it will claim a worse resolution once every 9 tries (10% + 1% + 0.1% + …), that's the perfect breeding ground for Heisenbugs.

*If* this guessing approach is kept, please add tests to it. (See below.)